### DISCUSSION TITLE
More than 4 motors?

### DISCUSSION THREAD

**Discussion by cbradburne** [luni64/TeensyStep4#22] [General]
Correct me if I'm wrong, but I believe TS4 will only cover 4 synchronous motors, and that there was mention of re-writing the code to allow for more. I fully appreciate this is not a small task, but I wondered if this is something you were still looking in to?
As always, your time working on TeensyStep is very much appreciated.

**Comment by maxitech** [comment]
Hi! I encountered the same issue when trying to control more than 4 motors **simultaneously**. After some research and testing, I found a solution that worked for me: **adding additional timer modules to the TimerFactory**.
By adding more timer modules to the TimerFactory, you can extend the capacity. 

## How it works
The Teensy 4.x uses **TMR timers**, and each TMR module has **4 timer channels** available. By default, the library registers one module (typically `TMRModule<3>`) in `teensystep4.cpp`, which provides enough timers for **4 motors**. However, if you need to control more motors, you can add additional timer modules manually. For example: 

```cpp
TimerFactory::attachModule(new TMRModule<0>());
```

This allows up to 8 motors by combining TMRModule<0> and the default TMRModule<3>. 

**Note:**  I’ve successfully tested this setup with 6 motors, as I don’t currently have the hardware for more. Based on the architecture of the TMR modules, this approach should theoretically work with up to 8 motors when using 2 TMR modules. If anyone tests it with more motors, feel free to share your results!

## Useful Resource for Timer Modules on Teensy Boards

For more detailed information about TMR modules, including how the Teensy 4.x uses these timers, check out this excellent resource:
[luni64/TeensyTimerTool GitHub Wiki](https://github-wiki-see.page/m/luni64/TeensyTimerTool/wiki/Supported-Timers)

I hope this helps you or others that come along!

...M
